apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java

```diff
@@ -176,14 +179,34 @@ public API(PropertiesHolder propertiesHolder, UPnP upnp, JettyConnectorCreator j
     }
 
     public static String findWebUiDir() {
-        String dir = DirProvider.getBinDir() + File.separator + WEB_UI_DIR;
-        dir = dir + File.separator + "build";
-        File res = new File(dir);
-        if (!res.exists()) { //we are in develop IDE or tests
-            dir = DirProvider.getBinDir() + "/apl-exec/target/" + WEB_UI_DIR + "/build";
-            res = new File(dir);
+        final Path binDir = DirProvider.getBinDir();
+
+        final Path htmlStubPath =
+            binDir.resolve("bin").resolve("html-stub").toAbsolutePath();
+        if (!Files.exists(htmlStubPath)) {
+            log.error("Cannot find dir: {}. Gonna proceed without any html-stub.", htmlStubPath);
+            return htmlStubPath.toString();
+        }
+
+        final String webUIProperty = propertiesHolder.getStringProperty("apl.webUIDir");
+        Path webUiPath;
+        try {
+            webUiPath = binDir.resolve(webUIProperty);
+
+            if (!Files.exists(webUiPath)
+                || !Files.isDirectory(webUiPath)
+                || !Files.exists(Path.of(webUiPath.toString(), "index.html"))
```

Comment:
webUiPath.resolve("index.html") = Path.of(webUiPath.toString(), "index.html")

Reply:
index.html is used in `Path.of(webUiPath.toString(), "index.html")`